be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala

```diff
@@ -83,10 +114,17 @@ object ElectionValidator extends MessageDataContentValidator with EventValidator
 
         val laoId: Hash = rpcMessage.extractLaoId
 
+        val sender: PublicKey = message.sender
+        val channel: Channel = rpcMessage.getParamsChannel
+
         if (!validateTimestampStaleness(data.created_at)) {
           Right(validationError(s"stale 'created_at' timestamp (${data.created_at})"))
         } else if (laoId != data.lao) {
           Right(validationError("unexpected lao id"))
+        } else if (!validateOwner(sender, channel)){
+          Right(validationError(s"invalid sender $sender"))
+        } else if (!validateChannelType(ObjectType.ELECTION, channel)) {
+          Right(validationError(s"trying to send a CastVoteElection message on a wrong type of channel $channel"))
```

Comment:
Did you mean `"trying to send a EndElection message on a wrong type of channel $channel"`

Reply:
yeah forgot to change again